plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -142,5 +197,12 @@ def container
         containers.first
       end
     end
+
+    def sidecar_container
+      @sidecar ||= begin
+        containers = template.spec.template.try(:spec).try(:containers) || []
+        containers.map { |possible_container| return possible_container if possible_container.name == 'secret-sidecar' }
```

Comment:
also if nothing is found this would return the map ... missing a testcase ?

```Ruby
containers.detect { |possible_container| break possible_container if possible_container.name == 'secret-sidecar' }
```